src/dune_rules/merlin.ml

```diff
@@ -94,11 +95,20 @@ module Processed = struct
               (Sexp.List (List.map ~f:(fun s -> Sexp.Atom s) flags))
           ]
       in
-      match pp with
-      | None -> flags
-      | Some { flag; args } ->
+      let flags =
+        match pp with
+        | None -> flags
+        | Some { flag; args } ->
+          make_directive "FLG"
+            (Sexp.List [ Atom (Pp_kind.to_flag flag); Atom args ])
+          :: flags
+      in
+      match mode with
+      | `Ocaml -> flags
+      | `Melange ->
         make_directive "FLG"
-          (Sexp.List [ Atom (Pp_kind.to_flag flag); Atom args ])
+          (Sexp.List
+             [ Atom (Pp_kind.to_flag Ppx); Atom "melc -as-ppx -bs-jsx 3" ])
```

Comment:
Shouldn't the melange ppx come first?

Reply:
Isn't the merlin syntax something like `FLG -ppx melc -as-ppx -bs-jsx 3`? The first atom is what renders the `-ppx` flag next to `FLG`.